chapter07/chapter07.adoc

```diff
@@ -0,0 +1,248 @@
+Base metrics are exposed under the path `/metrics?scope=base`.
+
+- *Application Metrics* are specific to an application, they are defined by the developers of the MicroProfile applications themselves. These are custom metrics that are specific to the business logic or operational aspects of the application. Developers use annotations or programmatic APIs to create and register these metrics, tailoring them to monitor the performance and behavior of their application's unique functionalities. Application metrics enable developers to gain insights into the runtime characteristics of their application, such as the number of transactions processed, response times for specific endpoints, or the rate of specific business events. 
+Application metrics are exposed under the path `/metrics?scope=application`.
+
+- *Vendor Metrics* are specific to a particular vendor or technology. These metrics provide insights into the performance and behavior of the runtime's internal components and extensions. Since different MicroProfile implementations may offer additional features or optimize certain areas differently, vendor metrics can vary widely between runtimes. They allow runtime vendors to expose unique metrics that are relevant to their implementation, offering users the ability to monitor vendor-specific aspects of their applications.
+Application metrics are exposed under the path `/metrics?scope=vendor`.
+
+Besides the standard scopes above, MicroProfile Metrics also supports custom scopes. You can use custom scopes to group sets of metrics that you frequently expect to view together.
+
+Note: In version 5.x, base metrics have become optional. This allows for flexibility in environments where these metrics may not be necessary or where they can be sourced from alternative monitoring tools.
+
+=== Metric Registry
+
+The *MetricRegistry* component acts as a container for storing and managing metrics within an application. It provides a structured way to collect, organize, and access various types of metrics (e.g., counters, gauges, histograms, and timers) for monitoring the behavior and performance of applications. It offers a centralized repository where metrics can be created and retrieved. This allows applications to consistently monitor critical operational and performance statistics.
+
+==== Types of Metric Registries
+
+MicroProfile Metrics defines several types of registries, categorized by their scope:
```

Comment:
MicroProfile Metrics creates metric registries for each scope:

Reply:
Fixed with a244b0b787c58554fc3eb15950a1efff180cf052